File: grpc-service/src/main/java/com/comparison/hotel/grpc/model/Reservation.java
package com.comparison.hotel.grpc.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "reservation")
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "client_id")
    private Client client;

    @ManyToOne
    @JoinColumn(name = "chambre_id")
    private Chambre chambre;

    @Column(name = "date_debut")
    private LocalDate dateDebut;

    @Column(name = "date_fin")
    private LocalDate dateFin;

    private String preferences;
}

File: graphql-service/entity/Reservation.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne } from "typeorm"
import { Client } from "./Client"
import { Chambre } from "./Chambre"

@Entity()
export class Reservation {
    @PrimaryGeneratedColumn()
    id: number

    @ManyToOne(() => Client, (client) => client.reservations)
    client: Client

    @ManyToOne(() => Chambre, (chambre) => chambre.reservations)
    chambre: Chambre

    @Column({ type: 'date' })
    dateDebut: string

    @Column({ type: 'date' })
    dateFin: string

    @Column("text")
    preferences: string
}
